scripts/frontmatter/get-frontmatter.js

```diff
@@ -0,0 +1,93 @@
+const frontmatter = require('@github-docs/frontmatter');
+const { TYPES } = require('../constants');
+
+const GATSBY_CONTENT_TYPES = {
+  [TYPES.BASIC_PAGE]: 'page',
+  [TYPES.API_DOC]: 'apiDoc',
+  [TYPES.RELEASE_NOTE]: 'releaseNote',
+  [TYPES.RELEASE_NOTE_PLATFORM]: 'releaseNotePlatform',
+  [TYPES.TROUBLESHOOTING]: 'troubleshootingDoc',
+  [TYPES.WHATS_NEW]: 'nr1Announcement',
+};
+
+const GATSBY_TEMPLATE = {
+  [TYPES.BASIC_PAGE]: 'basicDoc',
+  [TYPES.API_DOC]: 'basicDoc',
+  [TYPES.RELEASE_NOTE]: 'basicDoc',
+  [TYPES.RELEASE_NOTE_PLATFORM]: 'basicDoc',
+  [TYPES.TROUBLESHOOTING]: 'basicDoc',
+  [TYPES.WHATS_NEW]: 'basicDoc',
+};
+
+const FRONTMATTER_TEMPLATE = {
+  [TYPES.BASIC_PAGE]: 'page',
+  [TYPES.API_DOC]: 'apiDoc',
+  [TYPES.RELEASE_NOTE]: 'releaseNote',
+  [TYPES.RELEASE_NOTE_PLATFORM]: 'releaseNotePlatform',
+  [TYPES.TROUBLESHOOTING]: 'troubleshooting',
+  [TYPES.WHATS_NEW]: 'whatsNew',
+};
+
+const getFrontmatter = (type, doc) => {
+  const defaultFrontmatter = {
+    title: doc.title.replace(':', '-'),
+    contentType: GATSBY_CONTENT_TYPES[type],
+    template: GATSBY_TEMPLATE[type],
+    topics: getTopics(doc),
+  };
+
+  return frontmatter.stringify('', {
+    ...defaultFrontmatter,
+    ...addCustomFrontmatter[FRONTMATTER_TEMPLATE[type]](doc),
+  });
+};
+
+const addCustomFrontmatter = {
+  apiDoc: (doc) => {
+    return;
+  },
+  page: (doc) => {
+    return {
+      japaneseVersion: doc.japaneseVersionExists === 'yes',
+    };
+  },
+  troubleshooting: (doc) => {
+    return {
+      japaneseVersion: doc.japaneseVersionExists === 'yes',
+    };
+  },
+  releaseNote: (doc) => {
+    return {
+      releaseDateTime: doc.releasedOn || '',
```

Comment:
Out of curiosity, do we have have the "time" this was released? Seems we care more about the date itself than the specific time on that date the release was created (perhaps I'm wrong and not understanding something obvious). If that is the case, would it make more sense to leave this as `releasedOn`, or use `releaseDate` instead? Thoughts?

Reply:
it's a good question. on the docs site it stores the UNIX timestamp and we display the date and time. the reason we even record and display the time is because it is used to sort the sidebar nav for release notes for APM agents. that was done because there are some rare exceptions when release version number shouldn't be used to sort. so i think the thought was to just carry over date and time in case we need it in Gatsby.

Reply:
I can rename the variable, was just following the sample mdx! Does anybody have any preference?